src/coreComponents/physicsSolvers/wavePropagation/ElasticWaveEquationSEMKernel.hpp

```diff
@@ -0,0 +1,740 @@
+          arrayView1d< real64 > const damping_z,
+          arrayView1d< real64 > const mass )
+  {
+    forAll< EXEC_POLICY >( size, [=] GEOSX_HOST_DEVICE ( localIndex const k )
+    {
+
+      constexpr localIndex numNodesPerElem = FE_TYPE::numNodes;
+      constexpr localIndex numQuadraturePointsPerElem = FE_TYPE::numQuadraturePoints;
+
+
+      real64 xLocal[ numNodesPerElem ][ 3 ];
+      for( localIndex a = 0; a < numNodesPerElem; ++a )
+      {
+        for( localIndex i = 0; i < 3; ++i )
+        {
+          xLocal[a][i] = X( elemsToNodes( k, a ), i );
+        }
+      }
+
+      real64 N[ numNodesPerElem ];
+      real64 gradN[ numNodesPerElem ][ 3 ];
+
+      for( localIndex q = 0; q < numQuadraturePointsPerElem; ++q )
+      {
+        FE_TYPE::calcN( q, N );
+        real64 const detJ = m_finiteElement.template getGradN< FE_TYPE >( k, q, xLocal, gradN );
+
+        for( localIndex a = 0; a < numNodesPerElem; ++a )
+        {
+          real64 const localIncrement = density[k] * detJ * N[a];
+          RAJA::atomicAdd< ATOMIC_POLICY >( &mass[elemsToNodes[k][a]], localIncrement );
+        }
+      }
+
+      for( localIndex kfe = 0; kfe < numFacesPerElem; ++kfe )
+      {
+        localIndex const iface = elemsToFaces[k][kfe];
+
+        if( facesDomainBoundaryIndicator[iface] == 1 && freeSurfaceFaceIndicator[iface] != 1 )
+        {
+          for( localIndex q=0; q<numQuadraturePointsPerElem; ++q )
+          {
+            FE_TYPE::calcN( q, N );
+            real64 const detJ = m_finiteElement.template getGradN< FE_TYPE >( k, q, xLocal, gradN );
+
+            real64 invJ[3][3]={{0}};
+            FE_TYPE::invJacobianTransformation( q, xLocal, invJ );
+
+            for( localIndex a=0; a < numNodesPerFace; ++a )
+            {
+              /// compute ds=||detJ*invJ*normalFace_{kfe}||
+              real64 tmp[3]={0};
+              real64 ds = 0.0;
+              for( localIndex b=0; b<3; ++b )
+              {
+                for( localIndex j = 0; j < 3; ++j )
+                {
+                  tmp[b] += invJ[j][b]*faceNormal[iface][j];
+                }
+                ds +=tmp[b]*tmp[b];
+              }
+              ds = std::sqrt( ds );
+
+              localIndex numNodeGl = facesToNodes[iface][a];
+
+              // Damping in x=xpos and x=xneg direction
+              if( faceNormal[iface][0] < 0.0 || faceNormal[iface][0] > 0.0 )
+              {
+                real64 const alpha_x = density[k] * velocityVp[k];
+                real64 const alpha_y = density[k] * velocityVs[k];
+                real64 const alpha_z = density[k] * velocityVs[k];
+                damping_x[numNodeGl] += alpha_x*detJ*ds*N[a];
+                damping_y[numNodeGl] += alpha_y*detJ*ds*N[a];
+                damping_z[numNodeGl] += alpha_z*detJ*ds*N[a];
+              }
+              // Damping in y=ypos and y=yneg direction
+              if( faceNormal[iface][1] < 0.0 || faceNormal[iface][1] > 0.0 )
```

Comment:
Could it happen that the normal components are not _exactly_ 0?  In that case the behavior of this code could be indetermined. Would something like 
`
alpha_x = density[k] * (velocityVp[k]*abs(faceNormal[iface][0])+velocityVs[k]*sqrt(faceNormal[iface][1]*faceNormal[iface][1]*+faceNormal[iface][2]*faceNormal[iface][2]*))
`
be more appropriate?